src/App.js

```diff
@@ -14,11 +15,52 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    selectedGood: '',
+  }
+
+  clickHandler = (good) => {
+    this.setState({
+      selectedGood: good || '',
+    });
+  }
```

Comment:
you can create separate method `removeSelection` in which you would directly call `this.setState({ selected: '' })`